.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseTimeZoneParser.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using DateObject = System.DateTime;
+using Microsoft.Recognizers.Definitions.English;
+using Microsoft.Recognizers.Text.Number;
+
+namespace Microsoft.Recognizers.Text.DateTime
+{
+    public class BaseTimeZoneParser : IDateTimeParser
+    {
+        public static readonly string ParserName = Constants.SYS_DATETIME_TIMEZONE; //"TimeZone";
+
+        public BaseTimeZoneParser()
+        {
+        }
+
+        public ParseResult Parse(ExtractResult result)
+        {
+            return this.Parse(result, DateObject.Now);
+        }
+
+        // GetMinutes compute minutes shift from UTC 0 from matched numbers in texts. e.g. "-4:30" -> -270; "8"-> 480; "+8"-> 480
+        public int ComputeMinutes(string utcOffset)
+        {
+            if (utcOffset.Length == 0)
+            {
+                return Constants.InvalidOffsetValue;
+            }
+
+            int sign = 1; // earlier than utc, default value
+            if (utcOffset.StartsWith("+") || utcOffset.StartsWith("-"))
+            {
+                if (utcOffset.StartsWith("-"))
+                {
+                    sign = -1; // later than utc 0
+                }
+
+                utcOffset = utcOffset.Substring(1).Trim();
+            }
+
+            int hours = 0;
+            int minutes = 0;
+            if (utcOffset.Contains(":"))
+            {
+                List<string> splitParts = utcOffset.Split(':').ToList();
+                string f1 = splitParts[0];
```

Comment:
You can parse them directly. Assigning to vars called f1 and f2 doesn't help readability of the code.